plugins/techdocs-backend/src/search/DefaultTechDocsDocumentGenerator.ts

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { PluginEndpointDiscovery } from '@backstage/backend-common';
+import { Entity, RELATION_OWNED_BY } from '@backstage/catalog-model';
+import fetch from 'cross-fetch';
+import unescape from 'lodash/unescape';
+import { Logger } from 'winston';
+import pLimit from 'p-limit';
+import { Config } from '@backstage/config';
+import { CatalogApi, CatalogClient } from '@backstage/catalog-client';
+import { TechDocsDocument } from '@backstage/techdocs-common';
+
+interface MkSearchIndexDoc {
+  title: string;
+  text: string;
+  location: string;
+}
+
+export type TechDocsCollatorOptions = {
+  discovery: PluginEndpointDiscovery;
+  logger: Logger;
+  locationTemplate?: string;
+  catalogClient?: CatalogApi;
+  parallelismLimit?: number;
+  legacyPathCasing?: boolean;
+};
+
+type EntityInfo = {
+  name: string;
+  namespace: string;
+  kind: string;
+};
+
+export class DefaultTechDocsDocumentGenerator {
```

Comment:
If document generators is a concept we want to continue with moving forward, it would be nice updating the concept docs to explain the responsibilities of it.

Reply:
(Closing this convo in favor of discussing in the main thread. [See note](https://github.com/backstage/backstage/pull/7239#issuecomment-989652443))